File: app/src/main/java/de/bahnhoefe/deutschlands/bahnhofsfotos/model/PhotoStations.java
package de.bahnhoefe.deutschlands.bahnhofsfotos.model;

import java.util.ArrayList;
import java.util.List;

import lombok.Builder;
import lombok.NonNull;
import lombok.Value;

@Value
@Builder
public class PhotoStations {

    @NonNull
    String photoBaseUrl;

    @NonNull
    @Builder.Default
    List<PhotoLicense> licenses = new ArrayList<>();

    @NonNull
    @Builder.Default
    List<Photographer> photographers = new ArrayList<>();

    @NonNull
    @Builder.Default
    List<PhotoStation> stations = new ArrayList<>();

    public String getPhotographerUrl(final String photographer) {
        return photographers.stream()
                .filter(p -> p.getName().equals(photographer))
                .findAny()
                .map(p -> p.getUrl().toString())
                .orElse(null);
    }

    public String getLicenseName(final String licenseId) {
        return licenses.stream()
                .filter(license -> license.getId().equals(licenseId))
                .findAny()
                .map(PhotoLicense::getName)
                .orElse(null);
    }

    public String getLicenseUrl(final String licenseId) {
        return licenses.stream()
                .filter(license -> license.getId().equals(licenseId))
                .findAny()
                .map(l -> l.getUrl().toString())
                .orElse(null);
    }

}

File: app/src/main/kotlin/de/bahnhoefe/deutschlands/bahnhofsfotos/model/PhotoStations.kt
package de.bahnhoefe.deutschlands.bahnhofsfotos.model

data class PhotoStations @JvmOverloads constructor(
    var photoBaseUrl: String? = null,
    var licenses: List<PhotoLicense> = ArrayList(),
    var photographers: List<Photographer> = ArrayList(),
    var stations: List<PhotoStation> = ArrayList()
) {

    fun getPhotographerUrl(photographer: String): String? {
        return photographers
            .filter { p: Photographer -> p.name == photographer }
            .map { p: Photographer -> p.url.toString() }
            .firstOrNull()
    }

    fun getLicenseName(licenseId: String): String? {
        return licenses
            .filter { license: PhotoLicense -> license.id == licenseId }
            .map(PhotoLicense::name)
            .firstOrNull()
    }

    fun getLicenseUrl(licenseId: String): String? {
        return licenses
            .filter { license: PhotoLicense -> license.id == licenseId }
            .map { l: PhotoLicense -> l.url.toString() }
            .firstOrNull()
    }
}